Fix Bcc header missing with emails from conductor and test helpers

jduff opened the pull request:
### Summary

`Bcc` header was not being included with emails created by the actionmailbox conductor or the `create_inbound_email_from` test helpers. This change marks the `Bcc` header to be included when the email is encoded in both of those cases, using the `include_in_headers` features from the mail gem found [here](https://github.com/mikel/mail/blob/6bc16b4bce4fe280b19523c939b14a30e32a8ba4/lib/mail/fields/bcc_field.rb#L43)

jduff committed:
Fix Bcc header missing with emails from conductor and test helpers

kaspth commented on actionmailbox/app/controllers/rails/conductor/action_mailbox/inbound_emails_controller.rb:
```diff
@@ -28,6 +28,9 @@ def new_mail
 
       def create_inbound_email(mail)
+        # Bcc header is not encoded by default
+        mail[:bcc].include_in_headers = true if mail[:bcc]
```
Since this line isn't in `create_and_extract_message_id!` will that mean, Action Mailbox can't capture any BCC fields in emails?

kaspth replied:
Is there an option we can pass to `Mail.new` in `new_mail` instead? Besides `new_mail` seems like the more suitable place for this.

jduff replied:
Thanks for taking a look @kaspth!

> Since this line isn't in create_and_extract_message_id! will that mean Action Mailbox can't capture any BCC fields in emails?

The only reason this is a problem in these two places is because a `Mail` object is initialized and then serialized with `to_s` before passing it along, that doesn't occur with the other services afaik since the raw data forwarded by the other service is stored without creating a `Mail` object first (I only tested with Mailgun which included the `Bcc` header correctly).

> Is there an option we can pass to Mail.new in new_mail instead? Besides new_mail seems like the more suitable place for this.

The only reference I found to `include_in_headers` is within the `Bcc` field itself so I don't think it can be passed through `Mail.new`. Putting this in `new_mail` does seem suitable, I'll update the PR with that change.

jduff commented:
@kaspth are there any other changes that I should make to this before it can be merged?

rafaelfranca commented:
Hey @jduff can you rebase this? We applied https://github.com/rails/rails/commit/aab9f6734d1027ee5ee49c150f3e7b264e801a82 but I think we still need the change in the test helper.

jduff commented:
@rafaelfranca I just rebased and pushed it up. I think the commit you referenced updated tests so if you want me to drop the tests I added as well let me know.
